execution-frameworks/Invoke-AtomicRedTeam/Invoke-AtomicRedTeam.ps1

```diff
@@ -33,75 +30,127 @@ Github repo: https://github.com/redcanaryco/atomic-red-team
+        throw 'Unable to import the powershell-yaml module'
     }
 }
 
 function Get-AtomicTechnique {
-    [CmdletBinding()]
+    [CmdletBinding(DefaultParameterSetName = 'technique',
+        SupportsShouldProcess = $true,
+        PositionalBinding = $false,
+        ConfirmImpact = 'Medium')]
     Param(
+        [Parameter(Mandatory = $true,
+            Position = 0,
+            ValueFromPipelineByPropertyName = $true,
+            ParameterSetName = 'technique')]
+        [ValidateNotNull()]
+        [ValidateNotNullOrEmpty()]
         [string]
         $Path
     )
-    # Returns A HashTable For Each File Passed In
-    BEGIN { }
-    PROCESS {
+    
+    Write-Debug -Message "Getting atomic technique from $Path"
+
+    Process 
+    {
+        Write-Verbose -Message 'Attempting to convert files from yaml'
         foreach ($File in $Path) {
-            $parsedYaml = (ConvertFrom-Yaml (Get-Content $File -Raw ))
-            Write-Output $parsedYaml
+            if ($pscmdlet.ShouldProcess($File, 'Converting yaml file')) {
+                Write-Verbose -Message "Converting $File from Yaml"]
+                $parsedYaml = (ConvertFrom-Yaml (Get-Content $File -Raw))
+                Write-Output $parsedYaml
+            }
         }
     }
-    END { }
 }
 
 function Invoke-AtomicTest {
-    [CmdletBinding()]
+    [CmdletBinding(DefaultParameterSetName = 'technique',
+        SupportsShouldProcess = $true,
+        PositionalBinding = $false,
+        ConfirmImpact = 'Medium')]
     Param(
+        [Parameter(Mandatory = $true,
+            Position = 0,
+            ValueFromPipelineByPropertyName = $true,
+            ParameterSetName = 'technique')]
+        [ValidateNotNull()]
+        [ValidateNotNullOrEmpty()]
         [System.Collections.Hashtable]
         $AtomicTechnique,
 
+        [Parameter(Mandatory = $false,
+            Position = 1,
+            ValueFromPipelineByPropertyName = $true,
+            ParameterSetName = 'technique')]
         [switch]
         $GenerateOnly
     )
-    BEGIN {}
+    BEGIN { } # Intentionally left blank and can be removed
     PROCESS {
+        Write-Verbose -Message 'Attempting to run Atomic Techniques'
+
+        $techniqueCount = 0
         foreach ($Technique in $AtomicTechnique) {
-            $AtomicTest = $Technique.atomic_tests
+            $techniqueCount++
+            Write-Progress -Activity "Running $($Technique.display_name.ToString()) Technique" -Status 'Progress:' -PercentComplete ($techniqueCount / ($AtomicTechnique).Count * 100)
+            Write-Debug -Message "Gathering tests for Technique $Technique"
 
-            foreach ($Test in $AtomicTest) {
-                #Only Process Windows Tests For Now
+            $testCount = 0
+            foreach ($Test in $Technique.atomic_tests) {
+                $testCount++
+                Write-Progress -Activity 'Running Atomic Tests' -Status 'Progress:' -PercentComplete ($testCount / ($Technique.atomic_tests).Count * 100)
```

Comment:
This line's getting a bit long. I'd do the math as its own variable in a line above.

Reply:
This is resolved.